api/src/main/java/uk/gov/dvsa/mot/trade/read/core/TradeReadServiceDatabase.java

```diff
@@ -264,6 +278,54 @@ private DisplayMotTestItem mapMotTestToDisplayMotTestItem(MotTest motTest, Vehic
         return displayMotTestItem;
     }
 
+    private uk.gov.dvsa.mot.trade.api.Vehicle getLatestDvlaVehicleAndMapToTradeVehicle(List<DvlaVehicle> vehicles) {
+
+        uk.gov.dvsa.mot.trade.api.Vehicle tradeVehicle = null;
+
+        if (vehicles.isEmpty()) {
```

Comment:
Empty check here and null or empty check on line 289. Maybe do the null or empty check here and then no need for a check on line 289.